### PR TITLE
Tidy schema helper and add tests

### PR DESCRIPTION
**Pull request by benPearce1**
This PR does the following:
Removes the `util/schema.go` file and moves all schema functionality to the `schemas` package
Introduces an interface for all schema defintions: `EntitySchema`
Implements that schema on all resource/datasource schemas
Adds a test in the ensure that all attributes are coming from the appropriate source schema package
  e.g all attributes and nested attributes for a datasource come from `github.com/hashicorp/terraform-plugin-framework/datasource/schema` and vice-versa for resources

### CODE REVIEW

File: octopusdeploy_framework/datasource_spaces.go [hunk 1]
```diff
@@ -33,6 +33,7 @@ func (*spacesDataSource) Metadata(_ context.Context, _ datasource.MetadataReques
 }
 
 func (*spacesDataSource) Schema(_ context.Context, _ datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schemas.SpaceSchema{}.GetDatasourceSchema()
```
Comment (hnrkndrssn) [inline octopusdeploy_framework/datasource_spaces.go:36]: Should the assignment to `resp.Schema` below this be removed?

File: octopusdeploy_framework/resource_environment.go [hunk 2]
```diff
@@ -19,6 +19,8 @@ type environmentTypeResource struct {
 	*Config
 }
 
+var environmentSchema = schemas.EnvironmentSchema{}
```
Comment (hnrkndrssn) [inline octopusdeploy_framework/resource_environment.go:22]: Why did you change the convention used in earlier classes where you straight assign to `resp.Schema` without a local parameter?

File: octopusdeploy_framework/resource_lifecycle.go [hunk 3]
```diff
@@ -23,6 +23,7 @@ type lifecycleTypeResource struct {
 
 var _ resource.Resource = &lifecycleTypeResource{}
 var _ resource.ResourceWithImportState = &lifecycleTypeResource{}
+var lifecycleSchema = schemas.LifecycleSchema{}
```
Comment (hnrkndrssn) [inline octopusdeploy_framework/resource_lifecycle.go:26]: Same here?

File: octopusdeploy_framework/resource_nuget_feed.go [hunk 4]
```diff
@@ -26,16 +25,14 @@ func NewNugetFeedResource() resource.Resource {
 }
 
 var _ resource.ResourceWithImportState = &nugetFeedTypeResource{}
+var nugetFeedSchema = schemas.NugetFeedSchema{}
```
Comment (hnrkndrssn) [inline octopusdeploy_framework/resource_nuget_feed.go:28]: And another...

File: octopusdeploy_framework/resource_tenant_project.go [hunk 5]
```diff
@@ -52,7 +52,7 @@ func (t *tenantProjectResource) Metadata(ctx context.Context, req resource.Metad
 func (t *tenantProjectResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
 	resp.Schema = schema.Schema{
```
Comment (hnrkndrssn) [inline octopusdeploy_framework/resource_tenant_project.go:53]: Should this get its own schema class like the rest of the resources?

File: octopusdeploy_framework/resource_environment.go (hunk 2 above)
Comment (benPearce1) [reply octopusdeploy_framework/resource_environment.go:22]: Not sure, I think this was one of the earlier changes I made. I will go through and make this consistent

File: octopusdeploy_framework/resource_tenant_project.go (hunk 5 above)
Comment (benPearce1) [reply octopusdeploy_framework/resource_tenant_project.go:53]: Missed one. Thanks

**Review by hnrkndrssn — APPROVED** [review]

File: octopusdeploy_framework/datasource_spaces.go [hunk 6]
```diff
@@ -33,6 +33,7 @@ func (*spacesDataSource) Metadata(_ context.Context, _ datasource.MetadataReques
...
 	resp.Schema = schema.Schema{
```
[the remark this answers is not in the document]
Comment (benPearce1) [reply octopusdeploy_framework/datasource_spaces.go:37]: yes, it should

File: octopusdeploy_framework/resource_environment.go (hunk 2 above)
Comment (benPearce1) [reply octopusdeploy_framework/resource_environment.go:22]: fixed

File: octopusdeploy_framework/resource_lifecycle.go (hunk 3 above)
Comment (benPearce1) [reply octopusdeploy_framework/resource_lifecycle.go:26]: fixed

File: octopusdeploy_framework/resource_tenant_project.go (hunk 5 above)
Comment (benPearce1) [reply octopusdeploy_framework/resource_tenant_project.go]: fixed
